src/sentry/notifications/platform/sample.py

```diff
@@ -0,0 +1,284 @@
+        )
+
+
+@dataclass(frozen=True)
+class SecurityAlertData(NotificationData):
+    source = "security-monitoring"
+    alert_type: str
+    severity: str
+    project_name: str
+    description: str
+    affected_users: int
+    alert_url: str
+    acknowledge_url: str
+    escalate_url: str
+
+
+@template_registry.register(SecurityAlertData.source)
+class SecurityAlertNotificationTemplate(NotificationTemplate[SecurityAlertData]):
+    category = NotificationCategory.DEBUG
+
+    def render(self, data: SecurityAlertData) -> NotificationRenderedTemplate:
+        return NotificationRenderedTemplate(
+            subject=f"SECURITY ALERT: {data.alert_type} in {data.project_name}",
+            body=(
+                f"A {data.severity.upper()} security alert of type {data.alert_type} has been triggered for project {data.project_name}. "
+                f"The alert description: {data.description}. "
+                f"This security incident has affected {data.affected_users} users and requires immediate investigation and response."
+            ),
+            actions=[
+                NotificationRenderedAction(
+                    label="View Alert Details", link="https://example.com/security-alert"
+                ),
+                NotificationRenderedAction(
+                    label="Acknowledge", link="https://example.com/acknowledge"
+                ),
+                NotificationRenderedAction(
+                    label="Escalate to Security Team", link="https://example.com/escalate"
+                ),
+            ],
+            footer="This is a security alert requiring immediate attention.",
+        )
+
+    def render_example(self) -> NotificationRenderedTemplate:
+        return NotificationRenderedTemplate(
+            subject="SECURITY ALERT: Suspicious login pattern in my-app",
+            body=(
+                "A CRITICAL security alert of type Suspicious login pattern has been triggered for project my-app. "
+                "The alert description: Multiple failed login attempts detected from unusual locations. "
+                "This security incident has affected 23 users and requires immediate investigation and response."
+            ),
+            actions=[
+                NotificationRenderedAction(
+                    label="View Alert Details",
+                    link="https://example.com/security-alert",
+                ),
+                NotificationRenderedAction(
+                    label="Acknowledge",
+                    link="https://example.com/acknowledge",
+                ),
+                NotificationRenderedAction(
+                    label="Escalate to Security Team",
+                    link="https://example.com/escalate",
+                ),
+            ],
+            footer="This is a security alert requiring immediate attention.",
+        )
+
+
+@dataclass(frozen=True)
+class PerformanceAlertData(NotificationData):
```

Comment:
curious q, is there a benefit for using dataclass vs a typeddict ? or does it not matter?